Applications/Utils/MeshGeoTools/createIntermediateRasters.cpp

```diff
@@ -0,0 +1,140 @@
+/**
+ * \file
+ *
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include <memory>
+#include <string>
+
+#include <tclap/CmdLine.h>
+
+#include "BaseLib/BuildInfo.h"
+#include "BaseLib/FileTools.h"
+#include "Applications/ApplicationsLib/LogogSetup.h"
+#include "Applications/FileIO/AsciiRasterInterface.h"
+#include "GeoLib/Raster.h"
+
+int main(int argc, char* argv[])
+{
+    ApplicationsLib::LogogSetup logog_setup;
+
+    TCLAP::CmdLine cmd(
+        "Takes two DEMs located at the exact same spatial position (but at "
+        "different elevation) and calculates n raster DEMs located at "
+        "equidistant intervalls between them (i.e. for n=1, one new raster "
+        "located precisely in the middle will be created).\n\n"
+        "OpenGeoSys-6 software, version " +
+            BaseLib::BuildInfo::ogs_version +
+            ".\n"
+            "Copyright (c) 2012-2019, OpenGeoSys Community "
+            "(http://www.opengeosys.org)",
+        ' ', BaseLib::BuildInfo::ogs_version);
+    TCLAP::ValueArg<std::string> input1_arg(
+        "", "file1", "First DEM-raster file", true, "", "file1.asc");
+    cmd.add(input1_arg);
+    TCLAP::ValueArg<std::string> input2_arg(
+        "", "file2", "Second DEM-raster file", true, "", "file2.asc");
+    cmd.add(input2_arg);
+    TCLAP::ValueArg<std::string> output_arg("o", "output-file",
+                                            "Raster output file (*.asc)", true,
+                                            "", "output.asc");
+    cmd.add(output_arg);
+    TCLAP::ValueArg<std::size_t> number_arg(
+        "n", "number", "number of rasters to be calculated", false, 1, "int");
+    cmd.add(number_arg);
+
+    cmd.parse(argc, argv);
+
+    std::unique_ptr<GeoLib::Raster> dem1(
+        FileIO::AsciiRasterInterface::readRaster(input1_arg.getValue()));
+    std::unique_ptr<GeoLib::Raster> dem2(
+        FileIO::AsciiRasterInterface::readRaster(input2_arg.getValue()));
+
+    GeoLib::RasterHeader const h1 = dem1->getHeader();
```

Comment:
Could `h1` and `h2` be const references?

Reply:
:heavy_check_mark: